docs/source/user/parallel.rst

```diff
@@ -0,0 +1,97 @@
+perform other optimizations on (part) of a function. At the moment, this
+feature only works on CPUs.
+
+Some operations inside a user defined function, e.g., adding a scalar value to
+an array, are known to have parallel semantics.  A user program may contain
+many such operations and while each operation could be parallelized
+individually, such an approach often has lackluster performance due to poor
+cache behavior.  Instead, with auto-parallelization, Numba attempts to
+identify such operations in a user program, fuse adjacent ones together
+to form one or more kernels that are automatically run in parallel.
+The process is fully automated without modifications to the user program,
+which is in contrast to Numba's :func:`~numba.vectorize` or
+:func:`~numba.guvectorize` mechanism, where manual effort is required
+to create parallel kernels.
+
+
+Supported Operations
+====================
+
+In this section, we give a list of all the array operations that have
+parallel semantics and for which we attempt to parallelize.
+
+1. All numba array operations that are supported by :ref:`case-study-array-expressions`,
+   which include common arithmetic functions between Numpy arrays, and between
+   arrays and scalars, as well as Numpy ufuncs. They are often called
+   `element-wise` or `point-wise` array operations:
+
+    * unary operators: ``+`` ``-`` ``~``
+    * binary operators: ``+`` ``-`` ``*`` ``/`` ``/?`` ``%`` ``|`` ``>>`` ``^`` ``<<`` ``&`` ``**`` ``//``
+    * compare operators: ``==`` ``!=`` ``<`` ``<=`` ``>`` ``>=``
+    * :ref:`Numpy ufuncs <supported_ufuncs>` that are supported in :term:`nopython mode`.
+    * User defined :class:`~numba.DUFunc` through :func:`~numba.vectorize`.
+
+2. Numpy reduction function ``sum`` and ``prod``. Note that they have to be
+   written as ``numpy.sum(a)`` instead of ``a.sum()``.
+
+3. Numpy ``dot`` function between a matrix and a vector. When both inputs
+   are matrices, instead of parallelizing the matrix multiply, we choose to
+   leave it as a library call to Numpy's native implementation.
+
+4. Multi-dimensional arrays are also supported for the above operations
+   when operands have matching dimension and size. The full semantics of
+   Numpy broadcast between arrays with mixed dimensionality or size is
+   not supported, nor is the reduction across a selected dimension.
+
+Examples
+========
+
+In this section, we give an example of how this feature helps
+parallelize Logistic Regression::
+
+    @numba.jit(nopython=True, parallel=True)
+    def logistic_regression(Y,X,w,iterations):
+        for i in range(iterations):
+            w -= np.dot(((1.0 / (1.0 + np.exp(-Y * np.dot(X,w))) - 1.0) * Y),X)
+        return w
+
+We will not discuss details of the algorithm, but instead focus on how
+this program behaves with auto-parallelization:
+
+1. Input ``X`` is an ``N x D`` matrix. Input ``Y`` is a vector of size ``N``,
```

Comment:
Perhaps discuss these in order of appearance in the signature, or reorder the signature?